perses/rjmc/topology_proposal.py

```diff
@@ -300,38 +301,79 @@ class AtomMapper(object):
+        if it's not known, it can still be helpful as maps with the same score
+        are redundant --- i.e. a flipped phenyl ring.
+
+
+        Parameters
+        ----------
+        mol_A : oechem.oemol
+            old molecule
+        mol_B : oechem.oemol
+            new molecule
+        maps : list(dict)
+            list of maps to score
+
+        Returns
+        -------
+        list
+            list of the distance scores
+
+        """
+        coords_A = np.zeros(shape=(mol_A.NumAtoms(), 3))
+        for i in mol_A.GetCoords():
+            coords_A[i] = mol_A.GetCoords()[i]
+        coords_B = np.zeros(shape=(mol_B.NumAtoms(), 3))
+        for i in mol_B.GetCoords():
+            coords_B[i] = mol_B.GetCoords()[i]
+        from scipy.spatial.distance import cdist
+
+        all_to_all = cdist(coords_A, coords_B, 'euclidean')
+
+        mol_B_H = {x.GetIdx(): x.IsHydrogen() for x in mol_B.GetAtoms()}
+        all_scores = []
+        for M in maps:
+            map_score = 0
+            for atom in M:
+                if not mol_B_H[atom]:  # skip H's - only look at heavy atoms
+                    map_score += all_to_all[M[atom], atom]
+            all_scores.append(map_score)
+        return all_scores
+
     @staticmethod
-    def _get_mol_atom_map(current_oemol,
-                          proposed_oemol,
+    def _get_mol_atom_map(molA,
+                          molB,
                           atom_expr=None,
                           bond_expr=None,
                           map_strength='default',
                           allow_ring_breaking=True,
-                          matching_criterion = 'index',
+                          matching_criterion='index',
                           external_inttypes=False,
-                          return_all_maps=False,
-                          geometry='weak'):
-        """
-        Given two molecules, returns the mapping of atoms between them using the match with the greatest number of atoms
+                          map_strategy='core'):
+        """Find a suitable atom map between two molecules, according
+        to the atom_expr, bond_expr or map_strength
 
-        Arguments
-        ---------
-        current_oemol : openeye.oechem.oemol object
-             The current molecule in the sampler
-        proposed_oemol : openeye.oechem.oemol object
-             The proposed new molecule
-        atom_expr : int
-            Integer referring to atom mapping expression.
-            See `perses.utils.generate_expression`
-        bond_expr : int
-            Integer referring to bond mapping expression.
-            See `perses.utils.generate_expression`
-        map_strength : str default='default'
-            string corresponding to pre-set atom map strengths
-            ['default','weak','strong']
+        Parameters
+        ----------
+        molA : oechem.oemol
+            old molecule
```

Comment:
Must have geometry, right?

Reply:
Umm it depends what you mean by this. It needs to have some coordinates (wether they are good or not) for this to work. But we can make an oemol from a 2D input (like a smiles) and this will work fine.